### ISSUE TITLE
Direct link to audio

### ISSUE THREAD

**Issue by utya** [manbearwiz/youtube-dl-server#65]
Hi can i receive direct link to file like mp3 without donwload in to the dirrectory

**Comment by nbr23** [comment]
Hi @utya, I have a pending pull request (https://github.com/manbearwiz/youtube-dl-server/pull/59) that might answer your request. If you want to check it out until this gets merged into here, it is on the fork https://github.com/nbr23/youtube-dl-server
If you select metadata in the scroll down and put your URL, you will get a pop up with the various links connected to the youtube link.
If I misunderstood what you were looking for, please let me know.

**Comment by utya** [comment]
so i can download from finished and get link from metada, but in the metadata sometimes i can't find m4a audio

**Comment by nbr23** [comment]
The metadata fetching will present you with the files available on youtube (or other content hosts). In some cases, youtube-dl downloads one of the media available and extracts audio from it to create mp3 (or other) files. In that case there will be no direct link to the mp3 file, and you will need to download the video to get the audio.
For example, if you look at the metadata for https://www.youtube.com/watch?v=YkgkThdzX-8:

![screen](https://user-images.githubusercontent.com/1313493/84329095-45de3280-ab73-11ea-9e93-678d5b3c935b.png)
No mp3 is listed.
However, if you select `mp3` in the web ui, or use the following command:
```youtube-dl -x --audio-format mp3 https://www.youtube.com/watch?v=YkgkThdzX-8```
You will get an ffmpeg produced mp3.

Basically, if the format you want is not in the metadata, youtube-dl has to create it and needs to download the content locally first.
I hope this answers your questions.

**Comment by utya** [comment]
thnaks
